core/dbt/constants.py

```diff
@@ -1,9 +1,18 @@
+import os
+
 SECRET_ENV_PREFIX = "DBT_ENV_SECRET_"
 DEFAULT_ENV_PLACEHOLDER = "DBT_DEFAULT_PLACEHOLDER"
 METADATA_ENV_PREFIX = "DBT_ENV_CUSTOM_ENV_"
 
-MAXIMUM_SEED_SIZE = 1 * 1024 * 1024
-MAXIMUM_SEED_SIZE_NAME = "1MB"
+
+def get_max_seed_size():
+    mx = os.getenv("DBT_MAXIMUM_SEED_SIZE", "1")
+    return int(mx)
+
+
+DEFAULT_MAXIMUM_SEED_SIZE = 1 * 1024 * 1024
+MAXIMUM_SEED_SIZE = get_max_seed_size() * DEFAULT_MAXIMUM_SEED_SIZE
```

Comment:
I don't think we need both of these constants defined. `MAXIMUM_SEED_SIZE` should just be `get_max_seed_size() * 1024 * 1024`. The `get_max_seed_size()` function will return the default value of `1` if the user does not set/override.

Reply:
Thank you for checking @jtcohen6!

One reason for keeping `DEFAULT_MAXIMUM_SEED_SIZE` would be to be able to keep the logic of `load_seed_source_file()` in `read_files.py`. Since we do not convert the file from bytes in the new incremental logic we opted to continue computing the checksum with the "old" method for files smaller than 1 MiB so that we'd ensure backward compatibility with current manifests. This is since the checksum of a file encoded as anything other than UTF-8 will be different depending on if it gets UTF-8 encoded or is read as raw bytes.

The reason for not converting from bytes to UTF-8 string is performance and somewhat cleaner code but if we want to remove `DEFAULT_MAXIMUM_SEED_SIZE` maybe we should remove the elif and only compute the seed file hash incrementally instead. I see two options then:
1. Read the file as string and ensure UTF-8 encoding before hashing
2. Continue reading the raw bytes and accept that some projects with seed files that are not utf-8 might have a diff in their manifest deferral when upgrading to this version of dbt-core.

What's your thoughts on this?

Reply:
I've implemented this suggestion with removal of `DEFAULT_MAXIMUM_SEED_SIZE ` and went with alternative 1 above. The changes can be seen in https://github.com/dbt-labs/dbt-core/pull/7125/commits/53021d942863376f13a90f4dadacb4acfb82868f.